src/Modules/Manage/Dnn.PersonaBar.Users.Tests/AddRolesUnitTests.cs

```diff
@@ -0,0 +1,165 @@
+
+            // Act
+            var result = SetupCommandForAddRolesAndRun(userId.ToString(), "--roles", "Tester");
+
+            // Assert
+            Assert.IsFalse(result.IsError);
+        }
+
+        [TestCase]
+        public void Run_AddRolesWhenUserNotValid_ReturnErrorResponse()
+        {
+            // Arrange
+            var userId = 2;
+            UserInfo userInfo = null;
+
+            errorResultModel = new ConsoleErrorResultModel("Invalid userId");
+
+            _userValidatorMock
+                .Setup(u => u.ValidateUser(userId, portalSettings, null, out userInfo))
+                .Returns(errorResultModel);
+
+            // Act
+            var result = SetupCommandForAddRolesAndRun(userId.ToString());
+
+            // Assert
+            Assert.IsTrue(result.IsError);
+        }
+
+        [TestCase]
+        public void Run_AddRolesWhenRoleNotValid_ThrowsException()
+        {
+            // Arrange
+            var userId = 2;
+            var userInfo = GetUser(userId, true);
+
+            _userValidatorMock
+                .Setup(u => u.ValidateUser(userId, portalSettings, null, out userInfo))
+                .Returns(errorResultModel);
+
+            var userInfoList = new List<UserRoleInfo>(
+                    new[]
+                    {
+                        new UserRoleInfo
+                        {
+                            RoleID = 1,
+                            PortalID = testPortalId,
+                            IsPublic = true
+                        }
+                    }
+                );
+
+            var total = 1;
+            _usersControllerMock
+                .Setup(u => u.GetUserRoles(userInfo, "", out total, -1, -1))
+                .Returns(userInfoList);
+
+            var rolesList = new List<RoleInfo>(
+                    new[]
+                    {
+                        new RoleInfo
+                        {
+                            RoleID = 1,
+                            RoleName = "Tester"
+                        }
+                    }
+                );
+
+            _rolesControllerMock
+                .Setup(r => r.GetRolesByNames(portalSettings, -1, It.IsAny<IList<string>>()))
+                .Returns(rolesList);
+
+            // Act
+            TestDelegate ex = () =>
+            SetupCommandForAddRolesAndRun(userId.ToString(), "--roles", "Not Tester");
+
+            // Assert
+            Assert.Throws<Exception>(ex, "Should throw exception");
+        }
+
+        protected override ConsoleCommandBase DependencyInitializer()
+        {
+            return new AddRoles(_userValidatorMock.Object, _usersControllerMock.Object, _rolesControllerMock.Object);
+        }
+
+        private ConsoleResultModel SetupCommandForAddRolesAndRun(params string[] args)
+        {
+            args = new[] { "add-roles" }.Concat(args).ToArray();
+            command = SetupCommand(args);
+            return command.Run();
```

Comment:
I think it's a bit repetitive and unnecessary. I would shift more of this to the base class

```c#
private ConsoleResultModel SetupCommandForAddRolesAndRun(params string[] args)
{
   args = new[] { "add-roles" }.Concat(args).ToArray();
   return base.RunCommand(args);
}
```

where `base.RunCommand()` would replace `base.SetupCommand()`